packages/valory/skills/decision_maker_abci/redeem_info.py

```diff
@@ -124,14 +125,17 @@ def __hash__(self) -> int:
         """Custom hashing operator."""
         return hash(self.fpmm.condition.id) + hash(self.fpmm.question.id)
 
-    @property
-    def claimable_amount(self) -> int:
-        """Get the claimable amount of the current market."""
-        return int(self.outcomeTokenMarginalPrice * self.outcomeTokensTraded)
-
     @property
     def is_winning(self) -> bool:
         """Return whether the current position is winning."""
         our_answer = self.outcomeIndex
         correct_answer = self.fpmm.current_answer_index
         return our_answer == correct_answer
+
+    @property
+    def claimable_amount(self) -> int:
+        """Get the claimable amount of the current market."""
+        amount = int(self.outcomeTokenMarginalPrice * self.outcomeTokensTraded)
+        if self.is_winning:
+            return amount
+        return -amount
```

Comment:
Also why the claimable amount is -amount here? shouldn't it be 0?
What about markets declared invalid ?

Reply:
The amount is negative because the policy's reward should be negative as well:

https://github.com/valory-xyz/trader/blob/e075ac6ecbcf532a4fabbbc84d4879fd7652a470/packages/valory/skills/decision_maker_abci/behaviours/reedem.py#L117

https://github.com/valory-xyz/trader/blob/e075ac6ecbcf532a4fabbbc84d4879fd7652a470/packages/valory/skills/decision_maker_abci/behaviours/reedem.py#L123